### PR TITLE
DCOS-19579: add persistent volumes to multicontainer services

### PR DESCRIPTION
**Pull request by Poltergeist**
This introduces the support for persistent volumes in multi container services. 
For testing create a volume and select Persistent type add a name and a size.

In the JSON mode you should see something similar to this:

```JSON
...
"volumes": [
    {
      "persistent": {
        "size": 1
      },
      "name": "home"
    }
  ]
...
```

image

Closes DCOS-19579

### CODE REVIEW

**Comment by weblancaster** [comment]
@Poltergeist when I do the steps you described above I get some errors in the json editor. Please let me know if I'm not testing correctly.

screen shot 2017-12-04 at 1 58 08 pm

**Comment by ahoskins** [comment]
@Poltergeist I would also like more testing steps and expected results. I'm not exactly sure what is suppose to happen.

**Comment by Poltergeist** [comment]
Hey I changed the description to add some more steps. But regarding the image @weblancaster you  may have selected a single container service instead of a multi container service (pod) this feature has only been added to the last one. It existed quite some time already in single container services.

**Comment by Poltergeist** [comment]
Issue has been fixed, thanks for pointing it out.

**Comment by Poltergeist** [comment]
The Feature has not been merged to mesosphere/marathon yet so let's wait with merging the feature before.

**Comment by Poltergeist** [comment]
Alfred please test this.

**Comment by adam-mesos** [comment]
@Poltergeist https://github.com/mesosphere/marathon/pull/5813 just got merged to Marathon, and should make it into the Marathon bump for beta3. Anything else you're waiting on?

**Comment by Poltergeist** [comment]
Alfred please test this.

**Comment by Poltergeist** [comment]
alfred please test this

File: plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js [hunk 1]
```diff
@@ -169,6 +173,31 @@ class MultiContainerVolumesFormSection extends Component {
     }
   }
 
+  getLocalPersistentInput(volumes, key) {
+    if (volumes.type === VolumeConstants.type.localPersistent) {
```
Comment (weblancaster) [inline plugins/services/src/js/components/forms/MultiContainerVolumesFormSection.js]:
```
if (volumes.type !== VolumeConstants.type.localPersistent) {
   return;
}

// the rest of the code
```

**Comment by weblancaster** [comment]
Looks like there is an issue with this PR (screenshot)

screen shot 2017-12-20 at 1 28 41 pm

**Comment by Poltergeist** [comment]
Fixed the issue @weblancaster was experiencing

**Pull request merged**